pkg/controller/core/clusterqueue_controller.go

```diff
@@ -137,8 +137,12 @@ func (r *ClusterQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return ctrl.Result{}, nil
 }
 
-func (r *ClusterQueueReconciler) NotifyWorkloadUpdate(w *kueue.Workload) {
-	r.wlUpdateCh <- event.GenericEvent{Object: w}
+func (r *ClusterQueueReconciler) NotifyWorkloadUpdate(oldWl, newWl *kueue.Workload) {
+	if newWl != nil {
+		r.wlUpdateCh <- event.GenericEvent{Object: newWl}
+	} else if oldWl != nil {
```

Comment:
why else? what if the clusterqueue changed? In that case you would need to update both.